[Core] Phase out Print packets and added Countdown type to print json (#812)
ArchipelagoMW/Archipelago: docs/network protocol.md

| Name | Type | Notes |
| ---- | ---- | ----- |
| data | list\[[JSONMessagePart](#JSONMessagePart)\] | Type of this part of the message. |
| type | str | May be present to indicate the [PrintJsonType](#PrintJsonType) of this message. |
| receiving | int | Is present if type is Hint or ItemSend and marks the destination player's ID. |
| item | [NetworkItem](#NetworkItem) | Is present if type is Hint or ItemSend and marks the source player id, location id, item id and item flags. |
| found | bool | Is present if type is Hint, denotes whether the location hinted for was checked. |

> | countdown | int | Is present if type is Countdown, denotes the amount of seconds remaining on the countdown |

Reviewer: ```suggestion
| countdown | int | Is present if type is `Countdown`, denotes the amount of seconds remaining on the countdown. |
```

##### PrintJsonType

> PrintJsonType indicates the type of [PrintJson](#PrintJson) packet, different types can be handled differently by the client and can also contain different additional arguments. When receiving an unknown type the data's list\[[JSONMessagePart](#JSONMessagePart)\] should still be printed as normally. Currently known types:

Reviewer: ```suggestion
PrintJsonType indicates the type of [PrintJson](#PrintJson) packet, different types can be handled differently by the client and can also contain additional arguments. When receiving an unknown type the data's list\[[JSONMessagePart](#JSONMessagePart)\] should still be printed as normal. 

Currently defined types are:
```

| Type | Notes |
| ---- | ----- |

> | ItemSend | The message is in response to a player receiving an item |

Reviewer: ```suggestion
| ItemSend | The message is in response to a player receiving an item. |
```

> | Hint | The message is in response to a player hinting |

Reviewer: ```suggestion
| Hint | The message is in response to a player hinting. |
```

> | Countdown | The message contains information about the current server Countdown |

Reviewer: ```suggestion
| Countdown | The message contains information about the current server Countdown. |
```
